### ISSUE TITLE
v0.6.0 freezes on open

### ISSUE THREAD

**Issue by jovanovski** [droibit/flutter_custom_tabs#21]
Since 0.6.0, every time I try to launch a URL, the app and phone totally freeze. I have to lock and unlock the screen, and only then do I see the loaded custom tab.

Happens on multiple devices and versions of Android. Does not have this issue on 0.5.0

The code I use to launch
```
  void launchURL(String url) async {
    try {
      await launch(
        url,
        option: new CustomTabsOption(
            toolbarColor: mainColor,
            enableDefaultShare: true,
            enableUrlBarHiding: true,
            showPageTitle: true,
            animation: new CustomTabsAnimation.slideIn()),
      );
    } catch (e) {
      debugPrint(e.toString());
    }
  }
```

**Comment by jovanovski** [comment]
Correction, it's not only 0.6.0, previous versions are affected. Can't reproduce in debug/release mode, but once downloaded from the Play Store, this happens.

**Comment by Pyozer** [comment]
I've got the same issue.
I think it's due to R8 (Proguard).
Here is the logs:
```
11-06 15:36:17.838   889 10583 E WindowManager: Unhandled exception while laying out windows
11-06 15:36:17.838   889 10583 E WindowManager: java.lang.RuntimeException: Unknown animation name: x
11-06 15:36:17.838   889 10583 E WindowManager:         at android.view.animation.AnimationUtils.createAnimationFromXml(AnimationUtils.java:197)
11-06 15:36:17.838   889 10583 E WindowManager:         at android.view.animation.AnimationUtils.createAnimationFromXml(AnimationUtils.java:157)
11-06 15:36:17.838   889 10583 E WindowManager:         at android.view.animation.AnimationUtils.loadAnimation(AnimationUtils.java:138)
11-06 15:36:17.838   889 10583 E WindowManager:         at com.android.server.wm.AppTransition.loadAnimationRes(AppTransition.java:580)
11-06 15:36:17.838   889 10583 E WindowManager:         at com.android.server.wm.AppTransition.loadAnimation(AppTransition.java:1644)
11-06 15:36:17.838   889 10583 E WindowManager:         at com.android.server.wm.AppWindowToken.loadAnimation(AppWindowToken.java:1913)
11-06 15:36:17.838   889 10583 E WindowManager:         at com.android.server.wm.AppWindowToken.applyAnimationLocked(AppWindowToken.java:1823)
11-06 15:36:17.838   889 10583 E WindowManager:         at com.android.server.wm.AppWindowToken.setVisibility(AppWindowToken.java:450)
11-06 15:36:17.838   889 10583 E WindowManager:         at com.android.server.wm.WindowSurfacePlacer.handleClosingApps(WindowSurfacePlacer.java:509)
11-06 15:36:17.838   889 10583 E WindowManager:         at com.android.server.wm.WindowSurfacePlacer.handleAppTransitionReadyLocked(WindowSurfacePlacer.java:319)
11-06 15:36:17.838   889 10583 E WindowManager:         at com.android.server.wm.RootWindowContainer.performSurfacePlacement(RootWindowContainer.java:635)
11-06 15:36:17.838   889 10583 E WindowManager:         at com.android.server.wm.WindowSurfacePlacer.performSurfacePlacementLoop(WindowSurfacePlacer.java:207)
11-06 15:36:17.838   889 10583 E WindowManager:         at com.android.server.wm.WindowSurfacePlacer.performSurfacePlacement(WindowSurfacePlacer.java:155)
11-06 15:36:17.838   889 10583 E WindowManager:         at com.android.server.wm.WindowManagerService.relayoutWindow(WindowManagerService.java:2332)
11-06 15:36:17.838   889 10583 E WindowManager:         at com.android.server.wm.Session.relayout(Session.java:245)
11-06 15:36:17.838   889 10583 E WindowManager:         at android.view.IWindowSession$Stub.onTransact(IWindowSession.java:309)
11-06 15:36:17.838   889 10583 E WindowManager:         at com.android.server.wm.Session.onTransact(Session.java:165)
11-06 15:36:17.838   889 10583 E WindowManager:         at android.os.Binder.execTransact(Binder.java:752)
```

The plugin is doing this to init slideIn() animation:
```dart
const CustomTabsAnimation(
    startEnter: 'slide_in_right',
    startExit: 'slide_out_left',
    endEnter: 'android:anim/slide_in_left',
    endExit: 'android:anim/slide_out_right',
);
```
So may be, due to R8, the `android:anim/sldie_in_left` and `android:anim/slide_out_right` are not available anymore (misnamed or deleted by R8 ?)

**Comment by Pyozer** [comment]
Update:
I try to made a build without animations, like this:
```dart
    await CustomTabs.launch(
      url,
      option: CustomTabs.CustomTabsOption(
        toolbarColor: CustomColors.affMainBlue,
        enableDefaultShare: true,
        enableUrlBarHiding: true,
        showPageTitle: true,
        enableInstantApps: true,
        animation: CustomTabs.CustomTabsAnimation.slideIn(),
        extraCustomTabs: <String>[
          // ref. https://play.google.com/store/apps/details?id=org.mozilla.firefox
          'org.mozilla.firefox',
          // ref. https://play.google.com/store/apps/details?id=com.microsoft.emmx
          'com.microsoft.emmx',
        ],
      ),
    );
```

This fix the issue for me.

**Comment by XinyueZ** [comment]
Problem on Android:

The problem happens only when I release the app.
The debug version works always.

**Comment by Fleximex** [comment]
@Pyozer was on the right track. Though, I don't know if the issue specifically is related to Proguard/R8.

The problem is in the following code:

```
/// Create a built-in slide in animation.
factory CustomTabsAnimation.slideIn() {
  if (_slideIn == null) {
    _slideIn = const CustomTabsAnimation(
      startEnter: 'slide_in_right',
      startExit: 'slide_out_left',
      endEnter: 'android:anim/slide_in_left',
      endExit: 'android:anim/slide_out_right',
    );
  }
  return _slideIn;
}

/// Create a built-in fade animation.
factory CustomTabsAnimation.fade() {
  if (_fade == null) {
    _fade = const CustomTabsAnimation(
      startEnter: 'android:anim/fade_in',
      startExit: 'android:anim/fade_out',
      endEnter: 'android:anim/fade_in',
      endExit: 'android:anim/fade_out',
    );
  }
  return _fade;
}
```

**Comment by nws-jstorm** [comment]
For me the solution was to entirely remove the animation parameter.
Even `animation: CustomTabs.CustomTabsAnimation.slideIn(),` didnt make this work in release mode.

**Comment by woutervanwijk** [comment]
For clarity, you don't need to update the plugin, just launch with the animation-option, like this for example:
 await custom.launch(
          url,
          option: custom.CustomTabsOption(
            enableDefaultShare: true,
            animation: CustomTabsAnimation(
              startEnter: 'android:slide_in_right',
              startExit: 'android:slide_out_left',
              endEnter: 'android:anim/slide_in_left',
              endExit: 'android:anim/slide_out_right',
            ),
          ),
        );

**Comment by valen1321** [comment]
> Update:
> I try to made a build without animations, like this:
> 
> ```dart
>     await CustomTabs.launch(
>       url,
>       option: CustomTabs.CustomTabsOption(
>         toolbarColor: CustomColors.affMainBlue,
>         enableDefaultShare: true,
>         enableUrlBarHiding: true,
>         showPageTitle: true,
>         enableInstantApps: true,
>         animation: CustomTabs.CustomTabsAnimation.slideIn(),
>         extraCustomTabs: <String>[
>           // ref. https://play.google.com/store/apps/details?id=org.mozilla.firefox
>           'org.mozilla.firefox',
>           // ref. https://play.google.com/store/apps/details?id=com.microsoft.emmx
>           'com.microsoft.emmx',
>         ],
>       ),
>     );
> ```
> 
> This fix the issue for me.

that solve my issue thanks bro

**Comment by droibit** [comment]
I have merged #37 PR. 
In the next version I will release soon, `CustomTabsAnimation.slideIn()` will work fine in the release build.

Sorry for the late reply 🙇️.
